app/javascript/controllers/letters_calls_form_controller.js

```diff
@@ -0,0 +1,94 @@
+// Get the value of an element attribute as a number or a default value
+function getAttributeAsNumber(element, attributeName, defaultValue = 0) {
+	return parseFloat(element.getAttribute(attributeName)) || defaultValue;
+}
+
+// Calculate the total based on the values of letters and phone_calls inputs
+function calculateTotal(lettersValue, phoneCallsValue, unitPriceForLetters, unitPriceForPhoneCalls) {
+	const total = lettersValue * unitPriceForLetters + phoneCallsValue * unitPriceForPhoneCalls;
+	return parseFloat(total.toFixed(2)); // Format total to two decimal places
+}
+
+// Calculate the total with uplift
+function calculateTotalWithUplift(currentTotal, upliftPercentage) {
+	const upliftToAdd = currentTotal * (upliftPercentage / 100);
+	const totalWithUplift = (currentTotal + upliftToAdd);
+	return totalWithUplift;
+}
+
+// Update the DOM element with the provided total
+function updateDomElementTotal(totalAmount, element) {
+	totalAmount = parseFloat(totalAmount).toFixed(2);
+	element.innerHTML = '£' + totalAmount;
+}
+
+// Check if all necessary field values exist
+function checkIfAllFieldValueExists(lettersValue, phoneCallsValue) {
+	return lettersValue && phoneCallsValue;
+}
+
+// Handle change events on the form
+export function handleFormChange(event, form, totalElement, lettersElement, phoneCallsElement, upliftCheckboxElement, upliftPercentageElement) {
+	if ([lettersElement, phoneCallsElement, upliftPercentageElement, upliftCheckboxElement].includes(event.target)) {
+		// Reset uplift percentage value if uplift checkbox is unchecked
+		if (event.target === upliftCheckboxElement && !upliftCheckboxElement.checked) {
+			// Clear uplift percentage value
+			upliftPercentageElement.value = '';
+		}
+		// Get the values from the form
+		const lettersValue = parseFloat(lettersElement.value) || 0;
+		const phoneCallsValue = parseFloat(phoneCallsElement.value) || 0;
```

Comment:
Is it safe to default to 0 if the lettersElement value is null? Are there instances where this could indicate an error?